cmd/launcher/main.go

```diff
@@ -12,6 +12,14 @@ import (
 	"github.com/buildpack/lifecycle/cmd"
 )
 
+var (
+	launchDir string
+)
+
+func init() {
+	cmd.FlagLaunchDir(&launchDir)
```

Comment:
This flag is never parsed. I'm not sure if the launcher should have flags, given how it's used as an `ENTRYPOINT`. Maybe leave this unconfigurable for now?

Reply:
Hmm, looks like the launcher.go is using the launch dir value. Wouldn't things like this
```
	if err := os.Chdir(filepath.Join(l.DefaultLaunchDir, "app")); err != nil {
		return errors.Wrap(err, "change to app directory")
	}
```
break if it was configurable in builder but had a different default in launch. Maybe we should be added `-launch=<launch-dir>` with the correct value from build to the `ENTRYPOINT`

Reply:
Maybe write it into `/workspace/config` during the build?

Reply:
@sclevine Working on this I am realizing that `exporter` needs another input. Right now during export we are overloading the `<launch-dir>` concept. We need to know both the directory where the assets from which to generate layers live (`<layer>.toml`s, <layer-dir>s, etc.). AND we need to know at what path we want these layers to live in the exported image.

For example, when we call invoke `lifecycle.Exporter#export` from the `pack` cli, the assets actually live in a temp dir on the host even though we want `<launch-dir>` to be `/workspace` in the final image.

After getting some feedback from @dgodd and @jchesterpivotal, I propose we call the former concept `launchAssetDir` and add another flag to the reference implementation.

We needs to clear up these concepts so `<launch-dir>` can be truly configurable and so `exporter` can set that right value for `PACK_LAUNCH_DIR` on the exported image